src/components/SelectionManager.vue

```diff
@@ -246,8 +244,17 @@ export default defineComponent({
       this.$emit('delete', photos);
     },
 
-    deleteSelectedPhotosById(delIds: number[], selection: Selection) {
-      return this.deletePhotos(delIds.map((id) => selection.get(id)).filter((p): p is IPhoto => p !== undefined));
+    deleteSelectedPhotosById(delIds: number[], selection: SelectionProxy) {
+      let photosToDelete: IPhoto[] = []
+      for (const photo of selection.values()) {
+        if (photo === undefined) {
+          continue
+        }
+        if (delIds.indexOf(photo.fileid) > -1) {
+          photosToDelete.push(photo)
+        }
+      }
```

Comment:
1. Why not just `selection.values().filter( ... )`?
2. Can get a safer time complexity by making `delIds` a `Set` first